src/lib/y2users/users_module/reader.rb

```diff
@@ -0,0 +1,187 @@
+
+      # Reads the useradd defaults from Users module
+      #
+      # @return [UseraddConfig]
+      def read_useradd_defaults
+        users_map = Yast::Users.GetLoginDefaults
+        # notes: groups are deprecated and usrskell and create_mail_spool is not in users
+        y2users_map = {
+          group:             users_map["group"],
+          home:              users_map["home"],
+          umask:             users_map["umask"],
+          expiration:        users_map["expire"],
+          inactivity_period: users_map["inactive"]&.to_i,
+          shell:             users_map["shell"],
+          skel:              users_map["skel"]
+        }
+        UseraddConfig.new(y2users_map)
+      end
+
+      # Reads the Autlogin config from Autologin module
+      #
+      # @note Users reads and sets autologin directly and do not expose it
+      #   so it is easier to read it directly
+      #
+      # @return [LoginConfig]
+      def read_autologin
+        res = LoginConfig.new
+        res.autologin_user = if Yast::Autologin.user.empty?
+          nil
+        else
+          Yast::Autologin.user
+        end
+        res.passwordless = Yast::Autologin.pw_less
+
+        res
+      end
+
+      # Reads the users and groups
+      #
+      # @param config [Config]
+      def read_elements(config)
+        config.attach(groups)
+        config.attach(users(config))
+      end
+
+      # Returns the list of users
+      #
+      # @return [Array<User>] the collection of users
+      def users(config)
+        all_local = Yast::Users.GetUsers("uid", "local").values +
+          Yast::Users.GetUsers("uid", "system").values
+        all_local.map { |u| user(u, config) }
+      end
+
+      def groups
+        all_local = Yast::Users.GetGroups("cn", "local").values +
+          Yast::Users.GetGroups("cn", "system").values
+        all_local.map { |u| group(u) }
+      end
+
+      # Creates a {User} object based on the data structure of a Users user
+      #
+      # @param user_attrs [Hash] a user representation in the format used by Users
+      # @return [User]
+      def user(user_attrs, config)
+        user = User.new(user_attrs["uid"])
+
+        user.uid   = attr_value(:uidNumber, user_attrs)&.to_s
+        user.gid   = attr_value(:gidNumber, user_attrs)&.to_s
+        user.gid ||= config.groups.by_name(attr_value["groupname"])&.gid
+
+        user.shell = attr_value(:loginShell, user_attrs)
+        user.home  = attr_value(:homeDirectory, user_attrs)
+        user.gecos = [attr_value(:cn, user_attrs), *user_attrs["addit_data"].split(",")].compact
+        user.system = true if !user.uid && user_attrs["type"] == "system"
+
+        # set password only if specified, nil means not touch it
+        user.password = create_password(user_attrs) if user_attrs["userPassword"]
+
+        # set authorized keys only if set in users module
+        user.authorized_keys = user_attrs["authorized_keys"] if user_attrs["authorized_keys"]
+
```

Comment:
I miss some info for the user:

* Receive system mail?
* Home mode permissions
* Move home content?
* Create empty home?

Such attributes are not provided by `Y2Users::User` yet, but we are going to add them as part of https://github.com/yast/yast-users/pull/329.

Reply:
Not sure if two latest can be provided by the reader. Maybe I'm missing something.

Reply:
I plan to add it when User start having it, otherwise test suite failing.

Reply:
@dgdavid see testing data in tests. at least empty home aka no skeleton can be obtained from users.